packages/camera/camera_android_camerax/android/src/main/java/io/flutter/plugins/camerax/CameraAndroidCameraxPlugin.java

```diff
@@ -4,39 +4,67 @@
 
 package io.flutter.plugins.camerax;
 
+import android.content.Context;
 import androidx.annotation.NonNull;
 import io.flutter.embedding.engine.plugins.FlutterPlugin;
-import io.flutter.plugin.common.MethodCall;
-import io.flutter.plugin.common.MethodChannel;
-import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
-import io.flutter.plugin.common.MethodChannel.Result;
-
-/** CameraAndroidCameraxPlugin */
-public class CameraAndroidCameraxPlugin implements FlutterPlugin, MethodCallHandler {
-  /// The MethodChannel that will the communication between Flutter and native Android
-  ///
-  /// This local reference serves to register the plugin with the Flutter Engine and unregister it
-  /// when the Flutter Engine is detached from the Activity
-  private MethodChannel channel;
+import io.flutter.embedding.engine.plugins.activity.ActivityAware;
+import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding;
+import io.flutter.plugin.common.BinaryMessenger;
+
+/** Platform implementation of the camera_plugin implemented with the CameraX library. */
+public final class CameraAndroidCameraxPlugin implements FlutterPlugin, ActivityAware {
+  private InstanceManager instanceManager;
+  private FlutterPluginBinding pluginBinding;
+
+  /**
+   * Initialize this within the {@code #configureFlutterEngine} of a Flutter activity or fragment.
+   *
+   * <p>See {@code io.flutter.plugins.camera.MainActivity} for an example.
+   */
+  public CameraAndroidCameraxPlugin() {}
+
+  void setUp(BinaryMessenger binaryMessenger, Context context) {
+    // Set up instance manager.
+    instanceManager =
+        InstanceManager.open(
+            identifier -> {
+              new GeneratedCameraXLibrary.JavaObjectFlutterApi(binaryMessenger)
+                  .dispose(identifier, reply -> {});
+            });
+
+    // Set up Host APIs.
```

Comment:
nit: missing `GeneratedCameraXLibrary.JavaObjectHostApi.setup(...)`

Reply:
Oops! Ok I think I'm getting the pattern!